base/glibc-compatibility/memcpy/memcpy.h

```diff
@@ -214,3 +217,280 @@ static inline void * inline_memcpy(void * __restrict dst_, const void * __restri
 
     return ret;
 }
+
+}
+
+
+template <uint8_t SIZE>
+static inline void * MemcpyFixed(void * dest, const void * src)
+{
+    return detail::inline_memcpy(dest, src, SIZE);
+}
+
+/// The idea is taken from https://github.com/duckdb/duckdb/blob/master/src/include/duckdb/common/fast_mem.hpp
+static inline void * inline_memcpy(void * __restrict dest, const void * __restrict src, size_t size)
```

Comment:
But overlapping stores should not be worse.
The technique of overlapping stores is used by glibc, cosmopolitan libc, and folly.

Reply:
we'll still use this technique. the optimisation only tries to save smth by generating hopefully better instantiations of function with substituted `size`

Reply:
For small sizes, the compiler will completely replace `memcpy` with its own code:
https://godbolt.org/z/zKMThMvWb

Reply:
About 1% slowdown: https://s3.amazonaws.com/clickhouse-test-reports/44243/f48f4be976d969c6709ea7b49f2dcb6aa351b198/performance_comparison_[2/4]/report.html